core/templates/pages/exploration-editor-page/improvements-tab/services/improvement-modal.service.ts

```diff
@@ -411,11 +403,11 @@ angular.module('oppia').factory('ImprovementModalService', [
                   return;
                 }
                 $scope.messageSendingInProgress = true;
-                ThreadDataService.addNewMessage(
-                  threadId, tmpText, tmpStatus, function() {
+                ThreadDataService.addNewMessage(thread, tmpText, tmpStatus)
```

Comment:
You know, for some of these, I'm wondering if you should write addNewMessageAsync() and in general append Async() to the asynchronous methods. It makes reading and understanding the code a bit easier because it's clear what is done synchronously vs out-of-band.